dafni_cli/utils.py

```diff
@@ -70,4 +72,137 @@ def process_date_filter(date_str: str) -> str:
     Returns:
         str: Processed date str to YYYY-MM-DDT00:00:00
     """
-    return dt.strptime(date_str, "%d/%m/%Y").strftime("%Y-%m-%dT%H:%M:%S")
\ No newline at end of file
+    return dt.strptime(date_str, "%d/%m/%Y").strftime("%Y-%m-%dT%H:%M:%S")
+
+
+def check_key_in_dict(
+    input_dict: dict,
+    key: str,
+    nested_key: Optional[str] = None,
+    default: Optional[str] = "N/A",
+) -> Optional[Union[str, int]]:
+    """Utility function to check a nested dict for a given
+    key and nested key if applicable. If the keys exist, the
+    associated value is returned, otherwise the default value is returned.
+
+    Args:
+        input_dict (dict): dict to check in for keys
+        key (str): first key to check for
+        nested_key (Optional[str], optional): A nested key to check for. Defaults to None.
+        default (Optional[str], optional): default value if key(s) not found. Defaults to "N/A".
+
+    Returns:
+        Optional[Union[str, int]]: Value associated with given dict and key(s)
+    """
+    value = None
+    if isinstance(input_dict, dict):
+        if key in input_dict:
+            if nested_key and isinstance(input_dict[key], dict):
+                if nested_key in input_dict[key]:
+                    value = input_dict[key][nested_key]
+            elif not nested_key:
+                value = input_dict[key]
+    if value is not None:
+        return value
+    return default
+
+
+def process_dict_datetime(
+    input_dict: dict, key: str, nested_key: Optional[str] = None, default="N/A"
+) -> str:
+    """Utility function to check a nested dict for a given
+    key and nested key if applicable. If the keys exist, the
+    associated value is parsed to a datetime and then formatted to an
+    applicable datetime string, otherwise the default value is returned.
```

Comment:
This description doesn't mention that this only processes times that are in an ISO date format. Could this be added?

Reply:
updated